server/src/main/java/org/elasticsearch/action/admin/indices/datastream/GetDataStreamAction.java

```diff
@@ -119,11 +214,13 @@ public void writeTo(StreamOutput out) throws IOException {
 
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-            builder.startArray();
-            for (DataStream dataStream : dataStreams) {
+            builder.startObject();
```

Comment:
why was this top level `data_streams` field introduced? I don't think that this is needed?

Reply:
It did make it easier to verify in REST tests that the number of returned data streams was as expected with a `- length: { data_streams: 0 }` directive. I don't think the length directive otherwise provides a way to refer to the root level of a response.

Reply:
I think the `- length: ` directive should be improved to handle this? Now the response format is changed, because a test directive can't handle something. 

I also don't see length directive being used on the `data_streams` object, so maybe just keep using:

```
- match: { $body: [] }
``` 

?

Reply:
Those are fair points. 👍

Reply:
Went of the top array field to be consistent with component and composable templates eg. https://github.com/elastic/elasticsearch/blob/35159090b943d1044929d29464ff9fa718d1700d/server/src/main/java/org/elasticsearch/action/admin/indices/template/get/GetComponentTemplateAction.java#L146